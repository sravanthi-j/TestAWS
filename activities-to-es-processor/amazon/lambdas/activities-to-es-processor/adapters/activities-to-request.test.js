const expect = require('expect');
const Adapter = require('./activities-to-request').ActivitiesToRequestAdapter;
const ActivityEvent = require('./events/ActivityEvent');
const crypto = require('crypto');

function buildAdapter(getDate) {
  const awsKey = 'fakeKey';
  const awsKeyId = 'fakeKeyId';
  const awsToken = 'fakeToken';
  const endpoint = 'fake1.fakeRegion.fakeService.amazonaws.com';
  const elasticSearchIndex = 'fakeIndex';
  return new Adapter(awsKey, awsToken, awsKeyId, endpoint, elasticSearchIndex, getDate);
}
 

test('buildRequest changes array of Activities into an elastic search request message', () => {
  const getDate = () => new Date('2019-12-17T03:24:00Z');

  const events = [
    new ActivityEvent('123', new Date('2019-12-18T03:24:00Z'), 'Message 1'),
    new ActivityEvent('123', new Date('2019-12-19T03:24:00Z'), 'Message 2'),
  ];
  const adapter = buildAdapter(getDate);
  const result = adapter.buildRequest(getDate().toISOString().replace(/[:\-]|\.\d{3}/g, ''), events);

  expect(result).toMatchSnapshot();
});
 
test('buildAuthorizationHeader hashes and signs the request information for the authorization header', () => {
  const getDate = () => new Date('2019-12-17T03:24:00Z');
  const datetime = getDate().toISOString().replace(/[:\-]|\.\d{3}/g, '');

  const headers = {
    'Content-Type': 'application/json',
    'Host': 'fakeEndpoint',
    'Content-Length': 123,
    'X-Amz-Security-Token': 'fakeToken',
  };

  const adapter = buildAdapter(getDate);
  const result = adapter.buildAuthorizationHeader(datetime, 'POST', '/bulk', 'test body', headers);

  expect(result).toMatchSnapshot();
});

test('process builds the request and adds the authorization header', () => {
  const getDate = () => new Date('2019-12-17T03:24:00Z');

  const events = [
    new ActivityEvent('123', new Date('2019-12-18T03:24:00Z'), 'Message 1'),
    new ActivityEvent('123', new Date('2019-12-19T03:24:00Z'), 'Message 2'),
  ];
  const adapter = buildAdapter(getDate);
  const result = adapter.process(events);

  expect(result).toMatchSnapshot();
});