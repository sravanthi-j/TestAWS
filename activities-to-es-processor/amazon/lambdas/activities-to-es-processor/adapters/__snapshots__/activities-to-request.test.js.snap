// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildAuthorizationHeader hashes and signs the request information for the authorization header 1`] = `"AWS4-HMAC-SHA256 Credential=fakeKeyId/20191217/fakeRegion/fakeService/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-security-token, Signature=15b69bcf46a707566a6462dbc33f98eb6169fca41ac463a56525348d451df260"`;

exports[`buildRequest changes array of Activities into an elastic search request message 1`] = `
Object {
  "body": "{\\"index\\":{\\"_index\\":\\"fakeIndex\\",\\"_id\\":\\"123\\"}}
{\\"@id\\":\\"123\\",\\"@timestamp\\":\\"2019-12-18T03:24:00.000Z\\",\\"@message\\":\\"Message 1\\"}
{\\"index\\":{\\"_index\\":\\"fakeIndex\\",\\"_id\\":\\"123\\"}}
{\\"@id\\":\\"123\\",\\"@timestamp\\":\\"2019-12-19T03:24:00.000Z\\",\\"@message\\":\\"Message 2\\"}
",
  "headers": Object {
    "Content-Length": 244,
    "Content-Type": "application/json",
    "Host": "fake1.fakeRegion.fakeService.amazonaws.com",
    "X-Amz-Date": "20191217T032400Z",
    "X-Amz-Security-Token": "fakeToken",
  },
  "host": "fake1.fakeRegion.fakeService.amazonaws.com",
  "method": "POST",
  "path": "/_bulk",
}
`;

exports[`process builds the request and adds the authorization header 1`] = `
Object {
  "body": "{\\"index\\":{\\"_index\\":\\"fakeIndex\\",\\"_id\\":\\"123\\"}}
{\\"@id\\":\\"123\\",\\"@timestamp\\":\\"2019-12-18T03:24:00.000Z\\",\\"@message\\":\\"Message 1\\"}
{\\"index\\":{\\"_index\\":\\"fakeIndex\\",\\"_id\\":\\"123\\"}}
{\\"@id\\":\\"123\\",\\"@timestamp\\":\\"2019-12-19T03:24:00.000Z\\",\\"@message\\":\\"Message 2\\"}
",
  "headers": Object {
    "Authorization": "AWS4-HMAC-SHA256 Credential=fakeKeyId/20191217/fakeRegion/fakeService/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-security-token, Signature=1e41c0e24787f4126fdb40a0e573a362c3ce8cba9fd83449bd8cf479f04db86d",
    "Content-Length": 244,
    "Content-Type": "application/json",
    "Host": "fake1.fakeRegion.fakeService.amazonaws.com",
    "X-Amz-Date": "20191217T032400Z",
    "X-Amz-Security-Token": "fakeToken",
  },
  "host": "fake1.fakeRegion.fakeService.amazonaws.com",
  "method": "POST",
  "path": "/_bulk",
}
`;
